clippy_lints/src/explicit_write.rs

```diff
@@ -83,32 +84,57 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                 } else {
                     ""
                 };
+
+                // We need to remove the last trailing newline from the string because the
+                // underlying `fmt::write` function doesn't know wether `println!` or `print!` was
+                // used.
+                let mut write_output: String = write_output_string(write_args).unwrap();
```

Comment:
I'm not sure if this function can fail on some inputs, but if there are cases where this fails, it should probably fall back to not using a suggestion. Otherwise, maybe `expect()` would be better?